frontend/src/app/signin/page.tsx

```diff
@@ -0,0 +1,77 @@
+'use client'
+import React from "react";
+import signIn from "@/firebase/auth/signin";
+import { useRouter } from 'next/navigation'
+import { useAuthContext } from "@/context/authContext";
+
+function Page() {
+  const router = useRouter()
+  const { user } = useAuthContext();
+
+  if (!!user) {
+    router.push("/")
+  }
+
+  const [email, setEmail] = React.useState('')
+  const [password, setPassword] = React.useState('')
+  const [error, setError] = React.useState('')
+
+  const handleForm = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+
+    if (!email.match(/^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[A-Za-z]+$/)) {
+      setError('Niepoprawny email')
+      return
+    }
+
+    const { result, error } = await signIn(email, password);
```

Comment:
this should be within `try { } catch (...) `. Should this promise reject we have uncaught exception. Perhaps firebase api assures that this promise never rejects and sets `error` field in returned object, however i still belive we should try to catch it